test/parallel_api/numeric/numeric.ops/exclusive_scan.pass.cpp

```diff
@@ -48,46 +122,107 @@ test_with_usm(sycl::queue& q, const ::std::size_t count)
 
     // Run dpl::exclusive_scan algorithm on USM shared-device memory
     auto myPolicy = oneapi::dpl::execution::make_device_policy<
-        TestUtils::unique_kernel_name<class copy, TestUtils::uniq_kernel_index<alloc_type>()>>(q);
-    oneapi::dpl::exclusive_scan(myPolicy, d_idx, d_idx + count, d_val, 0);
+        TestUtils::unique_kernel_name<KernelName, TestUtils::uniq_kernel_index<alloc_type>()>>(q);
+    oneapi::dpl::exclusive_scan(myPolicy, d_idx, d_idx + count, d_val, init, binary_op);
 
     // Copy results from USM shared/device memory to host
     std::vector<int> h_val(count);
     dt_helper_h_val.retrieve_data(h_val.begin());
 
     // Check results
     std::vector<int> h_sval_expected(count);
-    exclusive_scan_serial(h_idx.begin(), h_idx.begin() + count, h_sval_expected.begin(), 0);
+    exclusive_scan_serial(h_idx.begin(), h_idx.begin() + count, h_sval_expected.begin(), init, binary_op);
 
     EXPECT_EQ_N(h_sval_expected.begin(), h_val.begin(), count, "wrong effect from exclusive_scan");
 }
 
-template <sycl::usm::alloc alloc_type>
+template <sycl::usm::alloc alloc_type, typename KernelName, typename _BinaryOp = oneapi::dpl::__internal::__pstl_plus>
 void
-test_with_usm(sycl::queue& q)
+test_with_usm(sycl::queue& q, _BinaryOp binary_op = _BinaryOp())
 {
     for (::std::size_t n = 0; n <= TestUtils::max_n; n = n <= 16 ? n + 1 : size_t(3.1415 * n))
     {
-        test_with_usm<alloc_type>(q, n);
+        test_with_usm<alloc_type, KernelName, _BinaryOp>(q, n);
     }
 }
 
 #endif // TEST_DPCPP_BACKEND_PRESENT
 
+template <typename _Tp>
+struct UserBinaryOperation
+{
+
```

Comment:
Not sure if this is a formatting issue, but there is an extra new line here,

Reply:
fixed